src/wireviz/wireviz.py

```diff
@@ -34,7 +35,11 @@ def parse(yaml_input: str, file_out: (str, Path) = None, return_types: (None, st
 
     yaml_data = yaml.safe_load(yaml_input)
 
-    harness = Harness()
+    # Assign default metadata.title here to avoid needing file_out in Metadata.__post_init__().
+    harness = Harness(
+        Metadata(**{'title': Path(file_out).stem, **yaml_data.get('metadata', {})}),
+        Options(**yaml_data.get('options', {})),
+    )
```

Comment:
```suggestion
    harness = Harness(
        metadata = yaml_data.get('metadata', {}),
        options = Options(**yaml_data.get('options', {})),
    )

    if not 'title' in harness.metadata:
        harness.metadata['title'] = Path(file_out).stem
```

A more pythonic and readable suggestion, assuming `Metadata` is kept as a normal `Dict`